tests/stress/stress_test.go

```diff
@@ -138,6 +138,36 @@ func TestSync_SingleBlockProducer(t *testing.T) {
 	}
 }
 
+func TestSync_MultipleEpoch(t *testing.T) {
+	numNodes = 3 // TODO: increase this when syncing improves
+	utils.SetLogLevel(log.LvlInfo)
+
+	// wait and start rest of nodes - if they all start at the same time the first round usually doesn't complete since
+	// all nodes vote for different blocks.
+	time.Sleep(time.Second * 5)
+	nodes, err := utils.InitializeAndStartNodes(t, numNodes, utils.GenesisDefault, utils.ConfigDefault)
+	require.NoError(t, err)
+
+	defer func() {
+		errList := utils.StopNodes(t, nodes)
+		require.Len(t, errList, 0)
+	}()
+
+	slotDuration, err := utils.SlotDuration(t, nodes[0])
+	require.NoError(t, err)
+	epochLength, err := utils.EpochLength(t, nodes[0])
+	require.NoError(t, err)
+
+	// Check state before wait
+
+	// Wait for epoch to pass
+	time.Sleep(time.Duration((slotDuration * epochLength) * 3))
+
+	// Just checking that everythings operating as expected
+	_, err = utils.SlotDuration(t, nodes[0])
+	require.NoError(t, err)
```

Comment:
this doesn't check that the nodes stayed synced - you need to use `compareBlocksByNumber` to make sure each node has the same block hash at each number

Reply:
Ahh rad was wondering what I could use